wgpu/src/api/buffer.rs

```diff
@@ -640,6 +652,28 @@ impl<'a> From<BufferSlice<'a>> for crate::BindingResource<'a> {
     }
 }
 
+fn range_overlaps(a: &Range<BufferAddress>, b: &Range<BufferAddress>) -> bool {
+    a.start < b.end && b.start < a.end
+}
+
+#[derive(Debug)]
+enum RangeKind {
+    Mutable,
+    Immutable,
+}
+
+impl RangeKind {
+    fn compatible(&self, other: &Self) -> bool {
```

Comment:
1. Making `RangeKind: Copy` and eliminating the `&` will make this code closer to optimal before optimization, by passing 2 bytes instead of references.
2. How about calling this function something more informative, like may_overlap()?

```suggestion
#[derive(Clone, Copy, Debug)]
enum RangeKind {
    Mutable,
    Immutable,
}

impl RangeKind {
    fn may_overlap(self, other: Self) -> bool {
```

Reply:
How about allowed_concurrently_with? Overlap just suggests ranges to me

Reply:
Sounds fine to me.